execution_engine/src/core/engine_state/genesis.rs

```diff
@@ -413,6 +406,898 @@ impl Distribution<ExecConfig> for Standard {
+                    .uref_address_generator
+                    .borrow_mut()
+                    .new_uref(AccessRights::READ_ADD_WRITE);
+
+                let round_seigniorage_rate: Ratio<U512> = {
+                    let (round_seigniorage_rate_numer, round_seigniorage_rate_denom) =
+                        self.exec_config.round_seigniorage_rate().into();
+                    Ratio::new(
+                        round_seigniorage_rate_numer.into(),
+                        round_seigniorage_rate_denom.into(),
+                    )
+                };
+
+                self.tracking_copy.borrow_mut().write(
+                    round_seigniorage_rate_uref.into(),
+                    StoredValue::CLValue(CLValue::from_t(round_seigniorage_rate).map_err(
+                        |_| GenesisError::CLValue(ARG_ROUND_SEIGNIORAGE_RATE.to_string()),
+                    )?),
+                );
+                round_seigniorage_rate_uref
+            };
+
+        let total_supply_uref = self
+            .uref_address_generator
+            .borrow_mut()
+            .new_uref(AccessRights::READ_ADD_WRITE);
+
+        let mut total_amount = U512::zero();
+
+        let purses: Vec<GenesisPurse> = self.create_purses()?;
+
+        let named_keys = {
+            let mut named_keys = NamedKeys::new();
+            named_keys.insert(
+                ROUND_SEIGNIORAGE_RATE_KEY.to_string(),
+                round_seigniorage_rate_uref.into(),
+            );
+
+            // associate purse urefs and balance urefs
+            for purse in purses.iter() {
+                match purse {
+                    GenesisPurse::ProofOfStake {
+                        purse_uref,
+                        balance_uref,
+                        amount,
+                    }
+                    | GenesisPurse::DelegatorReward {
+                        purse_uref,
+                        balance_uref,
+                        amount,
+                    }
+                    | GenesisPurse::ValidatorReward {
+                        purse_uref,
+                        balance_uref,
+                        amount,
+                    }
+                    | GenesisPurse::GenesisAccount {
+                        purse_uref,
+                        balance_uref,
+                        amount,
+                        ..
+                    }
+                    | GenesisPurse::GenesisValidator {
+                        purse_uref,
+                        balance_uref,
+                        amount,
+                        ..
+                    } => {
+                        total_amount = total_amount + amount;
+                        let purse_uref_name =
+                            purse_uref.remove_access_rights().to_formatted_string();
+                        let named_key = Key::URef(*balance_uref);
+                        named_keys.insert(purse_uref_name, named_key);
+                    }
+                }
```

Comment:
As mentioned above, we should be using `Runtime::call_host_mint` here instead of doing this manually.

Reply:
did this, via `exec_system_contract`